File: algorithms/Strings/Easy/ReverseString/VSDReverseString.java
public class Main{

    static void reverseString(String[] s){
        int i=0,j=s.length-1;
        while(i<j) swap(s,i++,j--);

    }
    static void swap(String[] s,int i,int j){
        String temp=s[i];
        s[i]=s[j];
        s[j]=temp;
    }
    public static void main(String[] args) {
        String[] s={"h","e","l","l","o"};
        reverseString(s);
        for(String a:s)System.out.print(a+" ");

    }
}

File: algorithms/Strings/Easy/ReverseString/VSDReverseString.cpp

#include<bits/stdc++.h>
using namespace std;

void swap(string s[],int i,int j){
    string temp=s[i];
    s[i]=s[j];
    s[j]=temp;
}

void reverseString(string s[],int n){
    int i=0,j=n-1;
    while(i<j)swap(s,i++,j--);
}


int main(){
    string s[]={"h","e","l","l","o"};
    int n=sizeof(s)/sizeof(s[0]);
    reverseString(s,n);
    for(string x:s)cout<<x<<" ";
    cout<<endl;
    return 0;
}